apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -1345,4 +1335,29 @@ defmodule ElixirLS.Debugger.Server do
     ref = Process.send_after(self(), :update_threads, 3000)
     %__MODULE__{state | update_threads_ref: ref}
   end
+
+  # Debugger Adapter Protocol stop reasons 'step' | 'breakpoint' | 'exception' | 'pause' | 'entry' | 'goto'
+  # | 'function breakpoint' | 'data breakpoint' | 'instruction breakpoint'
+  defp get_stop_reason(_state, :paused, _frames), do: "pause"
+  defp get_stop_reason(_state, :breakpoint_reached, []), do: "breakpoint"
+
+  defp get_stop_reason(state = %__MODULE__{}, :breakpoint_reached, [first_frame = %Frame{} | _]) do
+    file_breakpoints = state.breakpoints[first_frame.file] || []
+
+    function_breakpoints =
+      Map.new(state.function_breakpoints)[
+        {first_frame.module, first_frame.function, length(first_frame.args)}
+      ] || []
```

Comment:
split this up
```suggestion
    breakpoint_mfa = {first_frame.module, first_frame.function, length(first_frame.args)}
    function_breakpoints = state.function_breakpoints
    |> Map.new()
    |> Map.get(breakpoint_mfa, [])
```
Also, making a map here isn't really going to help performance if that's what you're looking for, recommend either making it a map or using `Enum.find`

Reply:
I was concerned with easier to use APIs. I'm not expecting many entries here. I'm going to change `state.function_breakpoints` to a map as this is not the first place where it needs to be accessed by key.